diff --git a/src/panicking.rs b/src/panicking.rs
index 1e6db9d..6fbd931 100644
--- a/src/panicking.rs
+++ b/src/panicking.rs
@@ -42,5 +42,5 @@ pub fn catch_unwind<E: Exception, R, F: FnOnce() -> R>(f: F) -> Result<R, Option
-        return if core::intrinsics::catch_unwind(do_call::<F, R>, data_ptr, do_catch::<E>) == 0 {
-            Ok(ManuallyDrop::into_inner(data.r))
-        } else {
+        return if core::intrinsics::catch_unwind(do_call::<F, R>, data_ptr, do_catch::<E>) {
             Err(ManuallyDrop::into_inner(data.p))
+        } else {
+            Ok(ManuallyDrop::into_inner(data.r))
         };
