diff --git a/src/addr.rs b/src/addr.rs
index 210ba16..6b1df13 100644
--- a/src/addr.rs
+++ b/src/addr.rs
@@ -416,0 +417,14 @@ impl Step for VirtAddr {
+
+    fn forward_overflowing(start: Self, count: usize) -> (Self, bool) {
+        match Self::forward_checked(start, count) {
+            Some(next) => (next, false),
+            None => (start, true),
+        }
+    }
+
+    fn backward_overflowing(start: Self, count: usize) -> (Self, bool) {
+        match Self::backward_checked(start, count) {
+            Some(next) => (next, false),
+            None => (start, true),
+        }
+    }
diff --git a/src/structures/paging/page.rs b/src/structures/paging/page.rs
index 911ecaa..367190e 100644
--- a/src/structures/paging/page.rs
+++ b/src/structures/paging/page.rs
@@ -299,0 +300,14 @@ impl<S: PageSize> Step for Page<S> {
+
+    fn forward_overflowing(start: Self, count: usize) -> (Self, bool) {
+        match Self::forward_checked(start, count) {
+            Some(next) => (next, false),
+            None => (start, true),
+        }
+    }
+
+    fn backward_overflowing(start: Self, count: usize) -> (Self, bool) {
+        match Self::backward_checked(start, count) {
+            Some(next) => (next, false),
+            None => (start, true),
+        }
+    }
